lib/config-generator.js

```diff
@@ -0,0 +1,278 @@
+`;
+        let exportContent = "";
+        let needCompatHelper = false;
+
+        if (this.answers.moduleType === "commonjs" || this.answers.moduleType === "script") {
+            exportContent += `  {files: ["**/*.js"], languageOptions: {sourceType: "${this.answers.moduleType}"}},\n`;
+        }
+
+        if (this.answers.env?.length > 0) {
+            this.result.devDependencies.push("globals");
+            importContent += "import globals from \"globals\";\n";
+            const envContent = {
+                browser: "globals: globals.browser",
+                node: "globals: globals.node",
+                "browser,node": "globals: {...globals.browser, ...globals.node}"
+            };
+
+            exportContent += `  {languageOptions: { ${envContent[this.answers.env.join(",")]} }},\n`;
+        }
+
+        if (this.answers.purpose === "syntax") {
+
+            // no need to install any plugin
+        } else if (this.answers.purpose === "problems") {
+            this.result.devDependencies.push("@eslint/js");
+            importContent += "import pluginJs from \"@eslint/js\";\n";
+            exportContent += "  pluginJs.configs.recommended,\n";
+        } else if (this.answers.purpose === "style") {
+            const styleguide = typeof this.answers.styleguide === "string"
+                ? { packageName: this.answers.styleguide, type: "flat" }
+                : this.answers.styleguide;
+
+            this.result.devDependencies.push(styleguide.packageName);
+
+            // install peer dependencies - it's needed for most eslintrc-style shared configs.
+            const peers = fetchPeerDependencies(styleguide.packageName);
+
+            if (peers !== null) {
+                this.result.devDependencies.push(...peers);
+            }
+
+            if (styleguide.type === "flat" || styleguide.type === void 0) {
+                importContent += `import styleGuide from "${styleguide.packageName}";\n`;
+                exportContent += "  ...[].concat(styleGuide),\n";
+            } else if (styleguide.type === "eslintrc") {
+                needCompatHelper = true;
+
+                const shorthandName = getShorthandName(styleguide.packageName, "eslint-config");
+
+                exportContent += `  ...compat.extends("${shorthandName}"),\n`;
+            }
+        }
+
+        if (this.answers.language === "typescript") {
+            this.result.devDependencies.push("typescript-eslint");
+            importContent += "import tseslint from \"typescript-eslint\";\n";
+            exportContent += "  ...tseslint.configs.recommended,\n";
+        }
+
+        if (this.answers.framework === "vue") {
+
+            this.result.devDependencies.push("eslint-plugin-vue");
+
+            importContent += "import pluginVue from \"eslint-plugin-vue\";\n";
+            exportContent += "  pluginVue.configs[\"flat/essential\"],\n";
```

Comment:
```suggestion
            exportContent += "  ...pluginVue.configs[\"flat/essential\"],\n";
```

This config is an array.

```
TypeError: Unexpected array.
    at flatTraverse (C:\projects\tmp\tmp\node_modules\@humanwhocodes\config-array\api.js:244:12)
    at flatTraverse.next (<anonymous>)
    at normalize (C:\projects\tmp\tmp\node_modules\@humanwhocodes\config-array\api.js:262:19)
    at async FlatConfigArray.normalize (C:\projects\tmp\tmp\node_modules\@humanwhocodes\config-array\api.js:677:30)
    at async calculateConfigArray (C:\projects\tmp\tmp\node_modules\eslint\lib\eslint\flat-eslint.js:438:5)
    at async FlatESLint.lintFiles (C:\projects\tmp\tmp\node_modules\eslint\lib\eslint\flat-eslint.js:745:25)
    at async Object.execute (C:\projects\tmp\tmp\node_modules\eslint\lib\cli.js:421:23)
    at async main (C:\projects\tmp\tmp\node_modules\eslint\bin\eslint.js:152:22)
```

Reply:
good catch! 👍